examples/example_server_forum.py

```diff
@@ -0,0 +1,155 @@
+    max_chars = 256
+    num_attributes = 2
+    max_chars -= (num_attributes*6+1) # "":"", and {} but last line has no comma
+    return "{\"user\":\"" + user + "\",\"msg\":\"" + msg + "\"}";
+
+# I did this in case any messages have newline characters (so I won't necessarily store comments as one line the posts file)
+def read_simple_json(json_str, start_idx):
+    return_dict = {}
+    in_quotes = False
+    in_esc = False
+    quote_start = None
+    param = None
+    seeker = start_idx
+    length = len(json_str)
+    if start_idx >= length or json_str[start_idx] != "{":
+        return None
+    while seeker < length:
+        cur_char = json_str[seeker]
+        if not in_quotes and not in_esc and cur_char == "}":
+            break
+        elif in_esc:
+            in_esc = False
+        elif cur_char == "\\":
+            in_esc = True
+        # handle quotes
+        elif cur_char == "\"":
+            if not in_quotes:
+                quote_start = seeker
+            if in_quotes:
+                if param == None and seeker+1 < length and json_str[seeker+1] == ":":
+                    param = json_str[quote_start+1:seeker]
+                else:
+                    return_dict[param] = json_str[quote_start+1:seeker]
+                    param = None
+                quote_start = None
+            in_quotes = not in_quotes
+        
+        seeker += 1
+    # end of while loop
+    return_dict['_length'] = seeker+1 - start_idx # how many chars to seek past in order to go past this json object
+    return return_dict
+
+# https://wiki.python.org/moin/BaseHttpServer
+class MyHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        print("")
+        print("GET for", self.client_address)
+        print("Path:", self.path)
+
+        output_txt = ""
+        data_dict = {}
+        try:
+            # params is everything after "?"
+            params = self.path[self.path.index("?")+1:]
+            data_dict = urllib.parse.parse_qs(params)
+        finally:
+            print("Request params:", data_dict)
+
+        try:
+            fd = open(posts_file, "r")
+            # If no user specified, get all comments.
+            if not 'user' in data_dict:
+                output_txt = fd.read()
+            # Fetch comments from specified user
+            else:
+                comment = ""
+                cnt = 0
+                fd = open(posts_file, "r")
+                while True:
+                    comment = fd.readline()
+                    if not comment:
+                        break
+                    cmt_json = read_simple_json(comment, 0)
+                    #print(cmt_json)
+                    if(cmt_json['user'] == data_dict['user'][0]):
+                        cnt += 1
+                        output_txt += comment
+                print("Number of matches:", cnt)
+                if(cnt == 0):
+                    output_txt = "This user has no comments."
+        except:
```

Comment:
I'd recommend moving this up to just after the open() call. That way it's clear that the except block will only be triggered if the file open call fails

Reply:
I believe I also had file reading in the try block in case it fails from a bad / non-open file. I'm looking it up now and it looks like I might be able to use "if not fd.closed" before the rest of the try block to move it out of the "try."
I may try and make this change on my next iteration.